hax/treemakers/corrections.py

```diff
@@ -207,26 +207,113 @@ def extract_data(self, event):
+            result['s2_xy_correction_bottom_' + algo] = (1.0 /
+                                                         self.corrections_handler.get_correction_from_map(
+                                                             "s2_xy_map", self.run_number,
+                                                             cvals, map_name='map_bottom'))
+
+            # Combine all the s2 corrections
+            result['cs2_' + algo] = \
+                s2.area * result['s2_lifetime_correction'] * result['s2_xy_correction_tot_' + algo]
+            result['cs2_top_' + algo] = \
+                s2.area * s2.area_fraction_top * \
+                result['s2_lifetime_correction'] * result['s2_xy_correction_top_' + algo]
+            result['cs2_bottom_' + algo] = \
+                s2.area * (1.0 - s2.area_fraction_top) * \
+                result['s2_lifetime_correction'] * result['s2_xy_correction_bottom_' + algo]
+
+            # Correct S1
+            cvals = [result['x_3d_' + algo], result['y_3d_' + algo], result['z_3d_' + algo]]
+            result['s1_xyz_correction_fdc_3d_' + algo] = \
+                (1 / self.corrections_handler.get_correction_from_map(
+                    "s1_lce_map_nn_fdc_3d", self.run_number, cvals))
+            result['cs1_no_field_corr_' + algo] = s1.area * result['s1_xyz_correction_fdc_3d_' + algo]
+
+            # Apply corrected LCE (light collection efficiency correction to s1, including field effects)
+            result['s1_xyz_true_correction_fdc_3d' + algo] = \
+                (1 / self.corrections_handler.get_correction_from_map(
+                    "s1_corrected_lce_map_nn_fdc_3d", self.run_number, cvals))
+            result['cs1_' + algo] = s1.area * result['s1_xyz_true_correction_fdc_3d' + algo]
+
+        # default cS1 and cS2 values
+        default_algo = 'nn_tf'
+        result['cs1'] = result['cs1_' + default_algo]
+        result['cs2'] = result['cs2_' + default_algo]
+        result['cs2_top'] = result['cs2_top_' + default_algo]
+        result['cs2_bottom'] = result['cs2_bottom_' + default_algo]
 
         return result
+
+class tfnn_position_reconstruction(object):
+    """
+    Class for loading and calculate tfnn position given area per channel
+    """
+    def __init__(self):
+        self.corrections_handler = CorrectionsHandler()
+        # We need to pull some stuff from the pax config
+        self.pax_config = load_configuration("XENON1T")
+        self.tpc_channels = self.pax_config['DEFAULT']['channels_in_detector']['tpc']
+        self.top_channels = self.pax_config['DEFAULT']['channels_top']
+
+        self.run_number = -1
+
+    def __call__(self, per_channel_area, run_number):
+        # per_channel_area as a list of channel area
+        # OR as an array of shape (#peaks, #channels w/o bad pmt removal)
+        # If we haven't loaded weights or it's another run reload nn
+        if ((not 'tfnn_weights' in self.__dict__.keys()) or
```

Comment:
__[Flake8]__ [E713] test for membership should be 'not in'

<sub>posted by Sider</sub>